test/e2e/targets/awssqs/main.go

```diff
@@ -0,0 +1,295 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awssqs
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"github.com/triggermesh/triggermesh/test/e2e/framework"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/apps"
+	e2esqs "github.com/triggermesh/triggermesh/test/e2e/framework/aws/sqs"
+	e2ece "github.com/triggermesh/triggermesh/test/e2e/framework/cloudevents"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/ducktypes"
+)
+
+/* This test suite requires:
+
+   - AWS credentials in whichever form (https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#hdr-Sessions_options_from_Shared_Config)
+   - The name of an AWS region exported in the environment as AWS_REGION
+*/
+
+var targetAPIVersion = schema.GroupVersion{
+	Group:   "targets.triggermesh.io",
+	Version: "v1alpha1",
+}
+
+const (
+	targetKind     = "AWSSQSTarget"
+	targetResource = "awssqstargets"
+)
+
+const awsAccessKeyIDSecretKey = "awsApiKey"
+const awsSecretAccessKeySecretKey = "awsApiSecret"
+
+var _ = Describe("AWS SQS target", func() {
+	f := framework.New("awssqstarget")
+
+	var ns string
+
+	var trgtClient dynamic.ResourceInterface
+
+	var queueURL string
+	var queueARN string
+	var awsCreds credentials.Value
+	var awsSecret *corev1.Secret
+
+	BeforeEach(func() {
+		ns = f.UniqueName
+
+		gvr := targetAPIVersion.WithResource(targetResource)
+		trgtClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+	})
+
+	Context("a target watches an existing queue", func() {
```

Comment:
A target doesn't really "watch" anything. I would keep the test context generic, something along the lines of "a target is deployed". Especially since the subcontext are describing quite well what we are testing ("When an event is sent to the target ... It puts a message onto the queue")